plugins/modules/vmware_guest_controller.py

```diff
@@ -361,6 +372,7 @@ def gather_disk_controller_facts(self):
                     controller_devicekey=device.key,
                     controller_unitnumber=device.unitNumber,
                     controller_disks_devicekey=device.device,
+                    controller_sharedbus=device.sharedBus,
```

Comment:
Personally, I find it a bit irritating to define a `bus_sharing` parameter and get `controller_sharedbus` in the result.

This is just a comment and not a requested change, mind. But when using this module, wouldn't you prefer a result that's closer to the parameter? `bus_sharing`, `controller_bus_sharing` or `controller_bussharing`? I'm not really sure about this myself but defining `bus_sharing` and getting `controller_sharedbus` feels a bit weird. What do you think?

@Akasurde @sky-joker @Tomorrow9 What's your opinion?

Reply:
I think you're right. Naming does matter and standardizing does make sense to me.

Reply:
I feel the same. Make parameter and output as same.

Reply:
I think I see where it came from:
line 375 uses the sharedBus `property` of `device`
Anyhow, I changed it.

Reply:
fixed in 1a37832

Reply:
@bushvin Also, please note that not all controllers have a `sharedBus` property. I think only [VirtualSCSIController](https://vdc-download.vmware.com/vmwb-repository/dcr-public/b50dcbbf-051d-4204-a3e7-e1b618c1e384/538cf2ec-b34f-4bae-a332-3820ef9e7773/vim.vm.device.VirtualSCSIController.html)s do. So you will have to check `isinstance(device, vim.vm.device.VirtualSCSIController)` or `hasattr(device, 'sharedBus')` before trying to access this. Otherwise, this line will [crash](https://86b1b8c44002f0715f04-1bfed7931d6589ee7c409888587537cf.ssl.cf5.rackcdn.com/878/e7ba6c144c9cfaeae916f89c2f06317402a6f7aa/check/ansible-test-cloud-integration-vcenter7_1esxi_without_nested-python36-stable210/c74d004/job-output.html#l56034).

Reply:
You're right.
I opted to detect the attribute in 59aa138
Probably only there in case of scsi controller, sbut you never know.

Reply:
> I opted to detect the attribute in [59aa138](https://github.com/ansible-collections/community.vmware/commit/59aa138eb01bfb692c0bbc9d4ca3705acb1a7527)
> Probably only there in case of scsi controller, but you never know.

Good decision. Even if SCSI controllers are the only ones with this property, this makes the code more future proof. After all, VMware might decide to introduce bus sharing for non-SCSI controllers in the future.